c7n/resources/cloudfront.py

```diff
@@ -181,9 +179,60 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('mismatch-s3-owner')
+class MismatchS3Owner(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront,
+       and verify whether owner is different from Cloudfront account owner.
```

Comment:
the more common case it will find is that the bucket does not exist not the owner is not the same as the account owner. ie. i'd switch owner in this filter name for origin.